src/corehost/src/main.cpp

```diff
@@ -8,30 +8,53 @@
 #include "utils.h"
 #include "coreclr.h"
 
-void get_tpafile_path(const pal::string_t& app_base, const pal::string_t& app_name, pal::string_t& tpapath)
+void get_tpafile_path(const pal::string_t& app_base, const pal::string_t& app_name, pal::string_t* tpapath)
 {
-    tpapath.reserve(app_base.length() + app_name.length() + 5);
+    tpapath->reserve(app_base.length() + app_name.length() + 5);
 
-    tpapath.append(app_base);
-    tpapath.push_back(DIR_SEPARATOR);
+    tpapath->append(app_base);
+    tpapath->push_back(DIR_SEPARATOR);
 
     // Remove the extension from the app_name
     auto ext_location = app_name.find_last_of('.');
     if (ext_location != std::string::npos)
     {
-        tpapath.append(app_name.substr(0, ext_location));
+        tpapath->append(app_name.substr(0, ext_location));
     }
     else
     {
-        tpapath.append(app_name);
+        tpapath->append(app_name);
     }
-    tpapath.append(_X(".deps"));
+    tpapath->append(_X(".deps"));
 }
 
-int run(arguments_t args, pal::string_t app_base, tpafile tpa)
+bool resolve_clr_path(const arguments_t& args, pal::string_t* clr_path)
 {
-    tpa.add_from_local_dir(app_base);
+    const pal::string_t* dirs[] = { &args.svc_dir, &args.app_dir, &args.home_dir };
```

Comment:
Please add a comment about the order of lookup since it reflects the policy (and is not random).